### PR TITLE
PIConGPU unit tests

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add units tets for PIConGPU
  - build the first test case to check particle shapes
- call unit tests in the CI
- remove unused trait `GetDataBoxType`
- cleanup include to avoid pulling full PIConGPU within shape tests
- cleanup includes in `simulation_types.hpp` and `simulation_defines.hpp`

### CODE REVIEW

File: include/picongpu/particles/Particles.hpp [hunk 1]
```diff
@@ -249,14 +249,6 @@ namespace picongpu
         }
     };
 
-    namespace traits
```
Comment (PrometheusPi) [inline include/picongpu/particles/Particles.hpp:252]: why is this removed?

File: include/picongpu/traits/GetDataBoxType.hpp
Comment (PrometheusPi) [inline include/picongpu/traits/GetDataBoxType.hpp:1]: why is this file removed?

File: share/ci/backendFlags.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+###################################################
+# translate PIConGPU backend names into CMake Flags
+###################################################
+
+get_backend_flags()
+{
+    backend_cfg=(${1//:/ })
+    num_options="${#backend_cfg[@]}"
```
Comment (PrometheusPi) [inline share/ci/backendFlags.sh:10]: why are you not using `backend_cfg[1]` as in the `result+=...` codes parts?

File: share/ci/run_picongpu_unit_tests.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+export code_DIR=$CI_PROJECT_DIR
+source $code_DIR/share/ci/backendFlags.sh
+
+# the default build type is Release
+# if neccesary, you can rerun the pipeline with another build type-> https://docs.gitlab.com/ee/ci/pipelines.html#manually-executing-pipelines
```
Comment (PrometheusPi) [inline share/ci/run_picongpu_unit_tests.sh]: This link does not seem to exist

File: share/ci/backendFlags.sh (hunk 2 above)
Comment (PrometheusPi) [reply share/ci/backendFlags.sh:10]: Okay - this is just a moved old method - it was so already in the old code

File: share/picongpu/unit/shape.cpp
Comment (PrometheusPi) [inline share/picongpu/unit/shape.cpp:1]: cool test, a bit more comments would help me better understand what is going on. (but not needed)

**Review by PrometheusPi — DISMISSED** [review]

File: include/picongpu/particles/Particles.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/Particles.hpp:252]: Since I was required to refactor the included order to avoid cyclic dependencies this shows up as unused stuff that introduces include issues. Moving this into a separate PR will introduce no advantage.

File: include/picongpu/traits/GetDataBoxType.hpp
Comment (psychocoderHPC) [reply include/picongpu/traits/GetDataBoxType.hpp:1]: since the last include was removed.

File: share/ci/backendFlags.sh (hunk 2 above)
Comment (psychocoderHPC) [reply share/ci/backendFlags.sh:10]: yes only moved the code to avoid copy-pasting it into all places where we use it.

File: share/ci/run_picongpu_unit_tests.sh (hunk 3 above)
Comment (psychocoderHPC) [reply share/ci/run_picongpu_unit_tests.sh]: I will remove the link

**Comment by psychocoderHPC** [comment]
I pushed the changes,

File: share/ci/run_picongpu_unit_tests.sh (hunk 3 above)
Comment (psychocoderHPC) [reply share/ci/run_picongpu_unit_tests.sh]: link is now removed

File: share/picongpu/unit/shape.cpp [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    }
+};
+
+
+/** Test a shape
+ *
+ * Evaluate the assignment shape at all grid points based on a random particle position.
+ * The sum of shape values must be 1.0.
+ *
+ * @tparam T_Shape assignment shape type, supports shape::ChargeAssignment and shape::ChargeAssignmentOnSUpport
```
Comment (PrometheusPi) [inline share/picongpu/unit/shape.cpp]:
```suggestion
 * @tparam T_Shape assignment shape type, supports shape::ChargeAssignment and shape::ChargeAssignmentOnSupport
```
typo

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
I am sorry, I found another typo

File: share/picongpu/unit/shape.cpp (hunk 4 above)
Comment (psychocoderHPC) [reply share/picongpu/unit/shape.cpp]: fixed

**Review by PrometheusPi — APPROVED** [review]
